estate/models/estate_property.py

```diff
@@ -57,3 +58,22 @@ def _onchange_garden(self):
         else:
             self.garden_area = False
             self.garden_orientation = False
+
+    def action_cancel(self):
+        for record in self:
+            if not record.state == 'sold':
+                record.state = 'canceled'
+            else:
+                raise UserError("You can't cancel a sold house")
```

Comment:
You could import `_` and wrap the string inside `_(...)` - so that you'll have it as a reference in case you look at this code later.